airtable/emails.js

```diff
@@ -0,0 +1,229 @@
+            details: delivery.getCellValue("Delivery Request Details"),
+            phone: delivery.getCellValue("Phone Number (D)"),
+            email: delivery.getCellValue("Email Optional (D)"),
+            instructions: delivery.getCellValue("Delivery Instructions"),
+            // we need to have a boolean value for each grocery store,
+            // in order to branch conditionally on the store in the email
+            grocery_store: {
+                shoprite: delivery.getCellValue("Grocery Store Name").name === "ShopRite, Little Falls",
+                golden_mango: delivery.getCellValue("Grocery Store Name").name === "Golden Mango, Paterson"
+            }
+        },
+        volunteer: {
+            name: volunteer.getCellValue("Name"),
+            phone: volunteer.getCellValue("Phone"),
+            email: volunteer.getCellValue("Email")
+        }
+    };
+}
+
+//==========================================================
+//==========================================================
+// Standard script begins here --
+// You shouldn't need to change anything below this line.
+//==========================================================
+
+// Define helper functions
+//==========================================================
+
+const matchedDeliveries = (await base.getTable("Deliveries").getView("Status: Matched").selectRecordsAsync()).records;
+const messagesTable = base.getTable("Messages")
+const volunteersTable = base.getTable("Volunteers")
+const deliveriesTable = base.getTable("Deliveries")
+const allMessages = (await messagesTable.selectRecordsAsync()).records;
+const allVolunteers = (await volunteersTable.selectRecordsAsync()).records;
+const allDeliveries = (await deliveriesTable.selectRecordsAsync()).records;
+
+// Compute a list of new queued messages to create, and return it.
+async function computeMessagesToCreate() {
+    let messagesToCreate = [];
+    for (const delivery of matchedDeliveries) {
+        const sendgridData = await getEmailInfoForDelivery(delivery);
+        // assemble a list of existing message types for this delivery
+        let existingMessageTypes = [];
+        const existingMessages = delivery.getCellValue("Messages");
+        if (existingMessages) {
+            existingMessageTypes = existingMessages.map(existingMessage =>
+                allMessages.find(m => m.id === existingMessage.id).getCellValue("Email type").name
+            );
+        } else {
+            existingMessageTypes = [];
+        }
+
+        // create a new message for the volunteer if necessary
+        if (existingMessageTypes.indexOf("Matched: to Volunteer") === -1) {
+            const volunteerEmail = sendgridData.volunteer.email;
+            if (volunteerEmail) {
+                messagesToCreate.push({
+                    fields: {
+                        "Delivery": [{ "id": delivery.id }],
+                        "Email type": { "name": "Matched: to Volunteer" },
```

Comment:
I think it's okay to keep these email types hardcoded for now, seem unlikely to vary too much across cities

Reply:
Yeah I didn't think that was an issue either.